### ISSUE TITLE
🚀 Auto apply from Run-Triggers

### ISSUE THREAD

**Issue by KamalAman** [hashicorp/hcp-terraform-operator#240]
### Description

Currently TFC states the following for Auto Apply on runs.

`
Plans queued due to run triggers from another workspace must always be manually applied.
`

Can you provide a way for the operator to automatically apply Planned runs as an optional configuration in addition to just depending on the built in TFC auto apply. 

This would provide some great benefits for ensuring the infrastructure is always applied an up to date.  

### Potential YAML Configuration
```yaml
  alwaysApplyRun: true
```

### References
https://developer.hashicorp.com/terraform/cloud-docs/workspaces/settings#apply-method

### Community Note

* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

**Comment by arybolovlev** [comment]
Hi @KamalAman,

Thank you for raising this question. This is not implemented because it might potentially create a loop in the case of misconfiguration. However, this might be revised in the future. Let's keep this FR open.

Thank you!

**Comment by arybolovlev** [comment]
Hi @KamalAman,

There is a new attribute `data.attributes.auto-apply-run-trigger` available in the Workspace API: https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspaces. I think this potentially can solve this request. It seems to be still in [beta](https://github.com/hashicorp/go-tfe/pull/798). I will keep an eye on it and we can return to this discussion once it is a more mature feature and available in TFE. 

Thanks!

**Comment by KamalAman** [comment]
It is in the UI under the workspaces settings as well, I think that should be used to this issue.
<img width="859" alt="image" src="https://github.com/hashicorp/terraform-cloud-operator/assets/2692357/9116b4b4-c181-4322-bbfd-2a5cce931659">

I believe the attribute would have to be added to the Workspace definition?

**Comment by arybolovlev** [comment]
Yeah, I think it would make sense to give it the same name and make it a boolean attribute:

```
autoApplyRunTriggers: < true | false >
```

**Comment by kGarciaSemarchy** [comment]
Hi @arybolovlev,

Any news on this ? This stop the automatic workflow that we use as we have a dependency between 2 workspaces managed by a trigger.

Thanks
